packages/contracts/src/EmailAccountRecovery.sol

```diff
@@ -124,45 +110,49 @@ abstract contract EmailAccountRecovery {
     /// @return address The computed address.
-    function computeEmailAuthAddress(
+    function computeProxyAddress(
         address recoveredAccount,
         bytes32 accountSalt
-    ) public view returns (address) {
-        // If on zksync, we use another logic to calculate create2 address.
-        if (block.chainid == 324 || block.chainid == 300) {
-            // return ZKSyncCreate2Factory(factory()).computeAddress(
-            //     accountSalt,
-            //     proxyBytecodeHash,
-            //     abi.encode(
-            //         emailAuthImplementation(),
-            //         abi.encodeCall(
-            //             EmailAuth.initialize,
-            //             (recoveredAccount, accountSalt, address(this))
-            //         )
-            //     )
-            // );
-        } else {
-            return
-                Create2.computeAddress(
-                    accountSalt,
-                    keccak256(
-                        abi.encodePacked(
-                            type(ERC1967Proxy).creationCode,
-                            abi.encode(
-                                emailAuthImplementation(),
-                                abi.encodeCall(
-                                    EmailAuth.initialize,
-                                    (
-                                        recoveredAccount,
-                                        accountSalt,
-                                        address(this)
-                                    )
+    ) public view virtual returns (address) {
+        return
+            Create2.computeAddress(
+                accountSalt,
+                keccak256(
+                    abi.encodePacked(
+                        type(ERC1967Proxy).creationCode,
+                        abi.encode(
+                            emailAuthImplementation(),
+                            abi.encodeCall(
+                                EmailAuth.initialize,
+                                (
+                                    recoveredAccount,
+                                    accountSalt,
+                                    address(this)
                                 )
                             )
                         )
                     )
-                );
-        }
+                )
+            );
+    }
+
+    /// @notice Deploys a new proxy contract for email authentication.
+    /// @dev This function uses the CREATE2 opcode to deploy a new ERC1967Proxy contract with a deterministic address.
+    /// @param recoveredAccount The address of the account to be recovered.
+    /// @param accountSalt A bytes32 salt value used to ensure the uniqueness of the deployed proxy address.
+    /// @return address The address of the newly deployed proxy contract.
+    function deployProxy(
```

Comment:
Also, could you rename it to `deployEmailAuthProxy`?
This function should be private/internal function.